src/Symfony/Component/Security/Http/RememberMe/PersistentRememberMeHandler.php

```diff
@@ -35,8 +35,45 @@ final class PersistentRememberMeHandler extends AbstractRememberMeHandler
     private TokenProviderInterface $tokenProvider;
     private ?TokenVerifierInterface $tokenVerifier;
 
-    public function __construct(TokenProviderInterface $tokenProvider, #[\SensitiveParameter] string $secret, UserProviderInterface $userProvider, RequestStack $requestStack, array $options, LoggerInterface $logger = null, TokenVerifierInterface $tokenVerifier = null)
+    /**
+     * @param UserProviderInterface       $userProvider
+     * @param RequestStack                $requestStack
+     * @param array                       $options
+     * @param LoggerInterface|null        $logger
+     * @param TokenVerifierInterface|null $tokenVerifier
+     */
+    public function __construct(TokenProviderInterface $tokenProvider, $userProvider, $requestStack, $options, $logger = null, $tokenVerifier = null)
```

Comment:
Shouldn't we keep the `#[\SensitiveParameter]` for the parameter `$userProvider` for the case there is injected a secret?

Reply:
I was wondering this myself. Would it hurt if no secret is passed an the user provider is omitted?

Reply:
Yes, your solution feels a bit to be a dirty hack. I know it's not so easy to deprecate such an argument. And just sth. comming into my mind, what about using the constructor with named argument `secret`?

Reply:
I am not sure I understand what solution you propose.

Reply:
I don't propose a solution, I proposed a maybe problem, i didn't test:
```
new PersistentRememberMeHandler (
tokenProvider: $tokenProvider,
secret: $secret, // <-- would this be a problem with your PR?
userProvider: $userProvider,
requestStack: $requestStack,
options: $options
)
```

Reply:
Ah, you are right about this. But named arguments are not covered by our BC promise: https://symfony.com/doc/current/contributing/code/bc.html#notes

Reply:
Maybe we can make it a bit more clear in the phpdoc or whereever? And I think for now we need the `#[\SensitiveParameter]` for `$userProvider`